File: steps_to_zero.c
#include <stdio.h>

int steps_to_zero(int num)
{
    int steps = 0;

    while (num)
    {
        if (num % 2)
            num--;
        else
            num = num >> 1;
    }

    return steps;
}

int main()
{
    printf("%d\n", 3547);
}

File: steps_to_zero.py
# Given a non-negative integer num, return the number of steps to reduce it to zero.
# If the current number is even, you have to divide it by 2,
# otherwise, you have to subtract 1 from it.

def count_steps(num):
    steps = 0

    while num != 0:
        if num % 2 == 1:
            num -= 1
        else:
            num /= 2

        steps += 1

    return steps

